ksqldb-execution/src/main/java/io/confluent/ksql/execution/expression/tree/LambdaVariable.java

```diff
@@ -22,7 +22,7 @@
 import java.util.Optional;
 
 @Immutable
-public class LambdaVariable extends Literal {
+public class LambdaVariable extends Expression {
```

Comment:
To make sure I understand - coercion util looks at expressions to fix type issues, so we need this to be an expression to fit with that. Is the scenario if we have something like `transform(arr, x => x + 2.3)` where arr is an array of decimals so x will be coerced as a lambda variable?

Reply:
I'd like to ask the same, for my own understanding as well :)

Reply:
Lambda Variables don't represent any values, they're basically like "local column references". If you look in coercion util at
```
    private static Expression convertToCommonType(
        final TypedExpression exp,
        final SqlType commonType
    )
```
Before I changed this to an expression, it was trying to convert the "value" of the lambda variable to whatever type was needed, which isn't the correct behavior, we should just be casting the lambda variable to the correct type.